### ISSUE TITLE
Server replies to PermissionQuery only once for each channel

### ISSUE THREAD

**Issue by garrappachc** [mumble-voip/mumble#5699]
### Description

I'm currently developing a bot that does some simple tasks for me; it essentially creates, removes and links channels (for now). According to the proto documentation, the `PermissionQuery` should be replied by the server whenever a client wants to query for channel permissions (Mumble.proto:465):
```proto
// Sent by the client when it wants permissions for a certain channel. Sent by
// the server when it replies to the query or wants the user to resync all
// channel permissions.
message PermissionQuery {
	// channel_id of the channel for which the permissions are queried.
	optional uint32 channel_id = 1;
	// Channel permissions.
	optional uint32 permissions = 2;
	// True if the client should drop its current permission information for all
	// channels.
	optional bool flush = 3 [default = false];
}
```
My bot queries for channel permissions in order to handle errors gracefully _before_ it attempts to create a new channel. However, the server replies with the `PermissionQuery` only once for each queried channel; any following queries are ignored. Any permission cache invalidation in the client will result in inability to ask for the channel permissions again.

### Steps to reproduce

1. Send `PermissionQuery` packet with `channelId = 0` _or_ send `PermissionQuery` packet with any other `channelId` more than once.

### Mumble version

1.4.230

### Mumble component

Server

### OS

Linux

### Reproducible?

Yes

### Additional information

My guess is that [this line](https://github.com/mumble-voip/mumble/blob/deace4ca92e7a6fb96fa235b0107bf2b72a21190/src/murmur/Server.cpp#L2044) is to be blamed. Invalidating this map when the server receives `PermissionQuery` packet from the client should result in this issue being fixed.

**Comment by Krzmbrzl** [comment]
As far as I see it, I think performing this check at all when the client explicitly asks for the permissions is nonsense.

Looking at https://github.com/mumble-voip/mumble/commit/7439bc4fe6b423ff0d36d3b1d529e474c14e9b01 it seems like this check mainly exists so that the automatic permission sending triggered by entering a channel does not flood a client with permission messages every time they enter a channel.
However, I really don't see why we'd need a check like this when the client explicitly asks for the permissions...

**Comment by garrappachc** [comment]
There is also another issue that comes with this one. Murmur re-uses IDs when creating new channels; for example when a new channel is created, then removed and another one is created afterwards, the new one will have the same ID as the removed one. This technically makes it impossible for the client to query for the permissions for the new channel, unless it caches permissions for inexisting channels.

**Comment by Krzmbrzl** [comment]
I would assume that the respective map is cleared when the channel is deleted. I haven't checked though...
